### PR TITLE
[thread-cert] refactor case 6.5.2 using pktverify

### PR DESCRIPTION
**Pull request by FiveDimensions**

### CODE REVIEW

File: tests/scripts/thread-cert/Cert_6_5_02_ChildResetReattach.py [hunk 1]
```diff
@@ -28,59 +28,223 @@
...
+        pv.verify_attached('ROUTER', 'LEADER')
+        pv.verify_attached('DUT', 'ROUTER', 'MTD')
+
+        if self.TOPOLOGY[MTD]['mode'] == 'rn':
+            # Step 4: DUT sends an MLE Child Update Request.
+            #         The following TLVs MUST be included in the Child Update
+            #         Request:
+            #             - Mode TLV
+            #             - Address Registration TLV (optional)
+            #         If the DUT is a SED, it MUST resume polling after sending
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/Cert_6_5_02_ChildResetReattach.py:145]: Check DUT SED resume polling?

Comment (FiveDimensions) [reply tests/scripts/thread-cert/Cert_6_5_02_ChildResetReattach.py]: added the check

File: tests/scripts/thread-cert/Cert_6_5_02_ChildResetReattach.py [hunk 2]
```diff
@@ -28,59 +28,238 @@
...
+import config
+from pktverify.consts import MLE_PARENT_REQUEST, MLE_PARENT_RESPONSE, MLE_CHILD_UPDATE_REQUEST, MLE_CHILD_UPDATE_RESPONSE, MLE_CHILD_ID_REQUEST, MLE_CHILD_ID_RESPONSE, ADDR_SOL_URI, SOURCE_ADDRESS_TLV, MODE_TLV, TIMEOUT_TLV, CHALLENGE_TLV, RESPONSE_TLV, LINK_LAYER_FRAME_COUNTER_TLV, MLE_FRAME_COUNTER_TLV, ROUTE64_TLV, ADDRESS16_TLV, LEADER_DATA_TLV, NETWORK_DATA_TLV, TLV_REQUEST_TLV, SCAN_MASK_TLV, CONNECTIVITY_TLV, LINK_MARGIN_TLV, VERSION_TLV, ADDRESS_REGISTRATION_TLV, NL_MAC_EXTENDED_ADDRESS_TLV, NL_RLOC16_TLV, NL_STATUS_TLV, NL_ROUTER_MASK_TLV, COAP_CODE_ACK
+from pktverify.packet_verifier import PacketVerifier
+from pktverify.null_field import nullField
 
 LEADER = 1
-ED = 2
+ROUTER = 2
+MTD = 3
+POLL_PERIOD = 3
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/Cert_6_5_02_ChildResetReattach.py:42]:
```suggestion
POLL_PERIOD = 3 # In seconds.
```
nit

File: tests/scripts/thread-cert/Cert_6_5_02_ChildResetReattach.py [hunk 3]
```diff
@@ -28,59 +28,238 @@
...
+        self.assertEqual(self.nodes[MTD].get_state(), 'child')
+
+        self.nodes[LEADER].clear_allowlist()
+        self.nodes[MTD].clear_allowlist()
+        self.nodes[ROUTER].stop()
 
+        self.nodes[MTD].reset()
+        self._setUpMTD()
+        if self.TOPOLOGY[MTD]['mode'] == '-':
+            self.nodes[MTD].set_pollperiod(POLL_PERIOD * 1000)
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/Cert_6_5_02_ChildResetReattach.py:114]: nit: move to _setUpMTD ?

File: tests/scripts/thread-cert/Cert_6_5_02_ChildResetReattach.py [hunk 4]
```diff
@@ -28,59 +28,238 @@
...
+                for i in (1, 4):
+                    pkts.filter_wpan_src64(DUT).\
+                        filter_wpan_dst64(ROUTER).\
+                        filter_mle_cmd(MLE_CHILD_UPDATE_REQUEST).\
+                        must_next
+                pkts.filter_wpan_src64(DUT).\
+                    filter_wpan_dst64(ROUTER).\
+                    filter_mle_cmd(MLE_CHILD_UPDATE_REQUEST).\
+                    filter(lambda p: p.sniff_time - _pkt.sniff_time <= POLL_PERIOD).\
+                    must_next
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/Cert_6_5_02_ChildResetReattach.py:171]:
```suggestion
                    must_next()
```

File: tests/scripts/thread-cert/Cert_6_5_02_ChildResetReattach.py [hunk 5]
```diff
@@ -28,59 +28,238 @@
...
+            must_next()
+        with pkts.save_index():
+            if self.TOPOLOGY[MTD]['mode'] == '-':
+                # filter another 3 MLE_CHILD_UPDATE_REQUEST with the upper one
+                # (total 4) sent out continuously in one poll
+                for i in (1, 4):
+                    pkts.filter_wpan_src64(DUT).\
+                        filter_wpan_dst64(ROUTER).\
+                        filter_mle_cmd(MLE_CHILD_UPDATE_REQUEST).\
+                        must_next
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/Cert_6_5_02_ChildResetReattach.py:166]:
```suggestion
                        must_next()
```

File: tests/scripts/thread-cert/Cert_6_5_02_ChildResetReattach.py [hunk 6]
```diff
@@ -28,59 +28,238 @@
...
+                # (total 4) sent out continuously in one poll
+                for i in (1, 4):
+                    pkts.filter_wpan_src64(DUT).\
+                        filter_wpan_dst64(ROUTER).\
+                        filter_mle_cmd(MLE_CHILD_UPDATE_REQUEST).\
+                        must_next()
+                pkts.filter_wpan_src64(DUT).\
+                    filter_wpan_dst64(ROUTER).\
+                    filter_mle_cmd(MLE_CHILD_UPDATE_REQUEST).\
+                    filter(lambda p: p.sniff_timestamp - _pkt.sniff_timestamp <= POLL_PERIOD).\
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/Cert_6_5_02_ChildResetReattach.py:170]:
```suggestion
                    filter(lambda p: p.sniff_timestamp - _pkt.sniff_timestamp <= POLL_PERIOD + 0.1).\
```
nit: we can add some allowance if necessary.

Comment (simonlingoogle) [inline tests/scripts/thread-cert/Cert_6_5_02_ChildResetReattach.py:170]: Should we check for the Data Request?

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1:

File: tests/scripts/thread-cert/Cert_6_5_02_ChildResetReattach.py (hunk 6 above)
Comment (FiveDimensions) [reply tests/scripts/thread-cert/Cert_6_5_02_ChildResetReattach.py]: yes

**Review by jwhui — APPROVED** [review]

**Pull request merged**